src/KDSoapServer/KDSoapServerObjectInterface.h

```diff
@@ -125,6 +127,17 @@ class KDSOAPSERVER_EXPORT KDSoapServerObjectInterface
      */
     virtual void processRequestWithPath(const KDSoapMessage &request, KDSoapMessage &response, const QByteArray &soapAction, const QString &path);
 
+    typedef QPair<QString/*name*/, QString/*value*/> HttpResponseHeaderItem;
+    typedef QList<HttpResponseHeaderItem> HttpResponseHeaderItems;
```

Comment:
Please use QVector. The pair/struct is bigger than a pointer, so QList has to allocate nodes on the heap, which is slow and wasteful.

Reply:
And QVector doesn't?  How does QVector allocate memory?

Reply:
QVector is like std::vector, it's a contiguous array. Especially if you resize() [or reserve()] upfront, a single allocation is needed, rather than one per 'node'.

Reply:
Ok, I figured it out now, thanks